corehq/apps/userreports/views.py

```diff
@@ -471,22 +474,25 @@ class ConfigureReport(ReportBuilderView):
     @use_nvd3
     @use_multiselect
     def dispatch(self, request, *args, **kwargs):
-        if self.existing_report:  # TODO: modify for existing registry report
+        if self.existing_report:
             self.source_type = get_source_type_from_report_config(self.existing_report)
             if self.source_type != DATA_SOURCE_TYPE_RAW:
                 self.source_id = self.existing_report.config.meta.build.source_id
                 self.app_id = self.existing_report.config.meta.build.app_id
                 self.app = Application.get(self.app_id) if self.app_id else None
+                self.registry_slug = self.existing_report.config.meta.build.registry_slug
             else:
                 self.source_id = self.existing_report.config_id
-                self.app_id = self.app = None
+                self.app_id = self.app = self.registry_slug = None
         else:
             self.registry_slug = self.request.GET.get('registry_slug', None)
             self.app_id = self.request.GET.get('application', None)
             if self.registry_slug:
-                self.source_type = 'case'
+                registry = DataRegistry.objects.get(slug=self.registry_slug)
+                DataRegistryHelper(self.domain, registry=registry).check_access(request.couch_user)
```

Comment:
```suggestion
                helper = DataRegistryHelper(self.domain, registry_slug=self.registry_slug)
                helper.check_access(request.couch_user)
```

(Note: `check_access` is expecting a case - maybe we should refactor it to just take the domain)

Reply:
looks like there are two `check_access` and `access_check`: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/registry/helper.py#L24 but I'm all for doing a refactor to use `access_check`

Reply:
well that's confusing, I'll make a note to try and improve the interface